homeassistant/components/sensibo/number.py

```diff
@@ -0,0 +1,130 @@
+        step=0.1,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Sensibo number platform."""
+
+    coordinator: SensiboDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        SensiboNumber(coordinator, device_id, description)
+        for device_id, device_data in coordinator.data.items()
+        for description in NUMBER_TYPES
+        if device_data["hvac_modes"] and device_data["temp"]
+    )
+
+
+class SensiboNumber(CoordinatorEntity, NumberEntity):
+    """Representation of a Sensibo numbers."""
+
+    coordinator: SensiboDataUpdateCoordinator
+    entity_description: SensiboNumberEntityDescription
+
+    def __init__(
+        self,
+        coordinator: SensiboDataUpdateCoordinator,
+        device_id: str,
+        entity_description: SensiboNumberEntityDescription,
+    ) -> None:
+        """Initiate Sensibo Number."""
+        super().__init__(coordinator)
+        self.entity_description = entity_description
+        self._device_id = device_id
+        self._client = coordinator.client
+        self._attr_unique_id = f"{device_id}-{entity_description.key}"
+        self._attr_name = (
+            f"{coordinator.data[device_id]['name']} {entity_description.name}"
+        )
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, coordinator.data[device_id]["id"])},
+            name=coordinator.data[device_id]["name"],
+            manufacturer="Sensibo",
+            configuration_url="https://home.sensibo.com/",
+            model=coordinator.data[device_id]["model"],
+            sw_version=coordinator.data[device_id]["fw_ver"],
+            hw_version=coordinator.data[device_id]["fw_type"],
+            suggested_area=coordinator.data[device_id]["name"],
+        )
+
+    @property
+    def value(self) -> float:
+        """Return the value from coordinator data."""
+        return self.coordinator.data[self._device_id][self.entity_description.key]
+
+    async def async_set_value(self, value: float) -> None:
+        """Set value for calibration."""
+        data = {self.entity_description.remote_key: value}
+        try:
+            async with async_timeout.timeout(TIMEOUT):
+                result = await self._client.async_set_calibration(
+                    self._device_id,
+                    data,
+                )
+        except SENSIBO_ERRORS as err:
+            raise HomeAssistantError(
+                f"Failed to set calibration for device {self.name} to Sensibo servers: {err}"
+            ) from err
+        LOGGER.debug("Result: %s", result)
+        if result["status"] == "success":
+            self.coordinator.data[self._device_id][self.entity_description.key] = value
+            self.async_write_ha_state()
+            return
+        raise HomeAssistantError(f"Could not set calibration for device {self.name}")
```

Comment:
This code looks very similar to `_async_set_ac_state_property` in `climate.py`.

In a future PR it would be good to make it a bit more DRY and reuse your new exception constant there.

Reply:
As also more service calls will be added that sounds like a good idea. Probably will make it when I'm going to make a base class for the entities. 
But then this PR needs to go in first.